### PR TITLE
T34712 add content node view

### PR DESCRIPTION
**Pull request by dylanmccall**
https://phabricator.endlessm.com/T34712

### CODE REVIEW

File: kolibri_dynamic_collections_plugin/assets/src/modules/collectionBase/handlers.js [hunk 1]
```diff
@@ -1,10 +1,7 @@
 export function showCollection(store) {
-  store.commit('CORE_SET_PAGE_LOADING', true);
-
   return Promise.all([store.dispatch('setAllChannelInfo')])
     .then(() => {
-      store.commit('CORE_SET_PAGE_LOADING', false);
-      store.commit('CORE_SET_ERROR', null);
+      // nothing to do...
```
Comment (dbnicholson) [inline kolibri_dynamic_collections_plugin/assets/src/modules/collectionBase/handlers.js]: I think just remove the `then`, right? The promise still needs to be resolved in the caller.

File: kolibri_dynamic_collections_plugin/assets/src/modules/collectionBase/index.js [hunk 2]
```diff
@@ -139,24 +139,31 @@ export default {
...
-      store.commit('SET_STATE', { ...store.state, externalTagsByNode, selectedNodeIdsByChannel });
+      store.commit('SET_STATE', { ...store.state, selectedNodeIdsByChannel });
     },
-    setExternalTagsForNode(store, { nodeId, tags }) {
+    removeSelectedNode(store, { channelId, nodeId }) {
+      const selectedNodeIdsByChannel = { ...store.state.selectedNodeIdsByChannel };
+
+      const channelNodeIds = new Set(selectedNodeIdsByChannel[channelId]);
+      channelNodeIds.delete(nodeId);
+      selectedNodeIdsByChannel[channelId] = Array.from(channelNodeIds);
```
Comment (dbnicholson) [inline kolibri_dynamic_collections_plugin/assets/src/modules/collectionBase/index.js]: Maybe it's just me, but it seems like constructing a set just to remove a single value from an array is excessive. I'd think the following would be more efficient:

```
const channelNodeIds = selectedNodeIdsByChannel[channelId].filter(e => e !== nodeId)
selectedNodeIdsByChannel[channelId] = channelNodeIds
```

Alternatively, why not just store the nodes as a set?

**Review by dbnicholson — COMMENTED** [review]
I left a couple nitpicky comments, but I'm not at all familiar enough with the collections editor or Vue to know if this is a good idea or not. In general it seems sane, though.

File: kolibri_dynamic_collections_plugin/assets/src/modules/collectionBase/index.js (hunk 2 above)
Comment (dylanmccall) [reply kolibri_dynamic_collections_plugin/assets/src/modules/collectionBase/index.js]: > Maybe it's just me, but it seems like constructing a set just to remove a single value from an array is excessive. I'd think the following would be more efficient:
> 
> ```
> const channelNodeIds = selectedNodeIdsByChannel[channelId].filter(e => e !== nodeId)
> selectedNodeIdsByChannel[channelId] = channelNodeIds
> ```

Yeah, that's a good point. Your one-liners there are simpler, even! I'll change that in a moment.

> Alternatively, why not just store the nodes as a set?

Alas, Vue's stores have this whole thing where data is mutable (distinct from Redux where everything is immutable) and the system magically handles mutations. But Map and Set aren't very well supported in this version. I don't really understand that in much detail, though. I already have three JavaScript state frameworks in my head and this one looks like all three of them and it freaks me out. So, uh, maybe I should just avoid Set altogether for consistency :)

**Comment by dylanmccall** [comment]
Thanks for the reviews! This inadvertently got an extra look from @pwithnall (in #14, which comes with all these commits _and more_ at half the price of competing brands). So I'll go ahead and merge this one :)

**Pull request merged**